snuba/utils/streams/kafka.py

```diff
@@ -147,19 +153,33 @@ def assignment_callback(
                         )
                     else:
                         raise ValueError("received unexpected offset")
-                self.__consumer.assign(assignment)
+
+                offsets: MutableMapping[TopicPartition, int] = {
+                    TopicPartition(i.topic, i.partition): i.offset for i in assignment
+                }
+                self.__seek(offsets)
             except Exception:
                 self.__state = KafkaConsumerState.ERROR
                 raise
 
-            if on_assign is not None:
-                on_assign([TopicPartition(i.topic, i.partition) for i in partitions])
+            try:
+                if on_assign is not None:
+                    on_assign(list(offsets.keys()))
+            finally:
+                self.__state = KafkaConsumerState.CONSUMING
 
         def revocation_callback(
             consumer: ConfluentConsumer, partitions: Sequence[ConfluentTopicPartition]
         ) -> None:
-            if on_revoke is not None:
-                on_revoke([TopicPartition(i.topic, i.partition) for i in partitions])
+            self.__state = KafkaConsumerState.REVOKING
+
+            streams = [TopicPartition(i.topic, i.partition) for i in partitions]
```

Comment:
don't we have to update self.__offsets in this method to remove the streams that are not assigned anymore ?

Reply:
I also managed to screw up and delete the test for this. 🤦‍♂